api/tests/Feature/ApplicantFilterTest.php

```diff
@@ -404,7 +403,7 @@ public function testFilterCanBeStoredAndRetrievedWithoutChangingResults()
                 'is_visible_minority' => $candidate->user->is_visible_minority,
                 'is_woman' => $candidate->user->is_woman,
                 'position_duration' => $candidate->user->position_duration,
-                'language_ability' => $candidate->user->language_ability,
+                'language_ability' => null, // user->language_ability is a computed field
```

Comment:
You can delete this line.

Reply:
no, I don't think I can without risking the test failing sometimes
the block as seen below attempts to match the filter to `$candidate`
what the accessor returns might not always be the appropriate result to filter with, so, to be safe and rather than randomly generate it in `factory()` I force it to null

the immediate problem case I can see is
```
lookingForEnglish = true
lookingForFrench = true
lookingForBilingual = false
```

in that case the accessor would assign BILINGUAL as best fit, but filtering for it wouldn't return the candidate

![image](https://user-images.githubusercontent.com/40485260/207411461-78a0f5a5-7349-4fac-9dd7-f11712970311.png)

Reply:
Hmmm... I see what you're saying, but this seems like a hack. I feel like there must be a better solution.

Reply:
Actually, wait. This is on ApplicantFilter, which _does_ still have the language_ability column in the database, and it needs to still work when using it to filter. We can't delete this line - we shouldn't modify it at all, in fact.

Reply:
Okay, so the problem with this is that when using the `language_ability` field to filter, we actually check the `looking_for_<language>` fields on the User. In that case, when generating a filter to match a specific user, we can use the `looking_for_<language>` fields instead of `user->language_ability`.

Reply:
alright, I'll try something else in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5101/commits/149a33af5e32b2f44d8fb50c4bbbfcb97d71c27d